app/services/charges/charge_models/percentage_service.rb

```diff
@@ -29,6 +57,9 @@ def compute_fixed_amount
 
       def free_units_value
         return 0 if last_running_total.zero?
+        if free_units_per_events.positive? && free_units_per_events < (aggregation_result.options[:running_total]&.count || 0)
```

Comment:
Layout/LineLength: Line is too long. [126/120] (https://rubystyle.guide#max-line-length)